.github/workflows/ci.yml

```diff
@@ -129,11 +131,101 @@ jobs:
+  ##############################################################################
+  rust-prebuild-matrix:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: 🐧 Linux
+            # All linux jobs must run the same ubuntu version to avoid Rust caching issues !
+            os: ubuntu-20.04
+          - name: 🍎 macOS
+            os: macos-12
+          - name: 🏁 Windows
+            os: windows-2022
+    name: '${{ matrix.name }}: 🦀 Rust Prebuild 🏗️'
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin v3.1.0
+
+      - uses: ./.github/actions/paths-filter
+        id: changes
+
+      - name: Check modified path that require rust-ci run
+        id: rust-changes
+        if: >-
+          steps.changes.outputs.rust == 'true'
+          || steps.changes.outputs.python == 'true'
+          || steps.changes.outputs.rust-ext == 'true'
+          || github.ref == 'refs/heads/master'
+        run: echo "run=true" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - uses: ./.github/actions/setup-rust
+        id: rust-cache
+        if: steps.rust-changes.outputs.run == 'true'
+        with:
+          version: ${{ env.rust-version }}
+          cache-key: integration
+          os: ${{ matrix.os }}
+        timeout-minutes: 10
+
+      # Pyo3 need additional configuration on MacOS to be manually build
+      # see https://pyo3.rs/v0.17.3/building_and_distribution.html#macos
+      - name: (🍎 macOS) Configure cargo for PyO3
+        if: >-
+          steps.rust-changes.outputs.run == 'true'
+          && steps.rust-cache.outputs.cache-hit != 'true'
+          && startsWith(matrix.os, 'macos-')
+        shell: bash
+        run: |
+          set -eux
+          set -o pipefail
+          mkdir -p .cargo
+          cat << EOF | tee .cargo/config.toml
+          [target.x86_64-apple-darwin]
+          rustflags = [
+            "-C", "link-arg=-undefined",
+            "-C", "link-arg=dynamic_lookup",
+          ]
+          EOF
+
+      - name: Fetch rust dependencies
+        if: >-
+          steps.rust-changes.outputs.run == 'true'
+          && steps.rust-cache.outputs.cache-hit != 'true'
+        shell: bash
+        run: |
+          find . -name Cargo.toml -not -path './oxidation/libparsec_python/*' -exec cargo fetch --verbose --locked --manifest-path \{\} \;
+        timeout-minutes: 20
+
+      - name: Build rust workspace
+        if: >-
+          steps.rust-changes.outputs.run == 'true'
+          && steps.rust-cache.outputs.cache-hit != 'true'
+        shell: bash
+        run: |
+          set -ex
+          cargo build \
+            --locked \
+            --workspace \
+            --profile ${{ env.CARGO_PROFILE }} \
+            --all-targets
+          cargo build \
+            --locked \
+            --workspace \
+            --profile ${{ env.CARGO_PROFILE }}
+        timeout-minutes: 40
```

Comment:
40? I would expect the build to be much quicker than that

Reply:
When I was testing, I see the macos runner that could take up to ~30 min to build `libparsec`

Reply:
oO